src/containers/search-bar-list/search-bar-list.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Typography from '@material-ui/core/Typography';
+
+import { useStyles } from './search-bar-list.styles';
+import { NOTHING_FOUND_MESSAGE } from '../../configs';
+import SearchBarListItem from './search-bar-list-item';
+import CircularLoadingBar from '../../components/circular-loading-bar';
+import { setProductsForSearchBar } from '../../redux/search-bar/search-bar.actions';
+
+const SearchBarList = () => {
+  const {
+    products,
+    language,
+    searchBarVisibility,
+    searchBarLoading
+  } = useSelector(({ SearchBar, Language }) => ({
+    products: SearchBar.list,
+    searchBarVisibility: SearchBar.visibility,
+    searchBarLoading: SearchBar.loading,
+    language: Language.language
+  }));
+
+  const styles = useStyles();
+  const dispatch = useDispatch();
+
+  useEffect(() => () => dispatch(setProductsForSearchBar([])), [dispatch]);
```

Comment:
Why you use this hook?

Reply:
It was necessary for clearing state when `Search Bar List `component are going to `unmount`. But you are right, no need to clear list.